File: backend/src/main/java/tom/projects/monkeylog/repository/workout/ProgramRepository.java
package tom.projects.monkeylog.repository.workout;

import org.springframework.data.jpa.repository.JpaRepository;
import tom.projects.monkeylog.model.workout.Program;

import java.util.List;
import java.util.UUID;

public interface ProgramRepository extends JpaRepository<Program, UUID> {
    List<Program> findAllByUserId(UUID userId);
}

File: backend/src/main/kotlin/tom/com/monkeylog/repository/workout/ProgramRepository.kt
package tom.com.monkeylog.repository.workout

import org.springframework.data.jpa.repository.JpaRepository
import tom.com.monkeylog.model.workout.Program
import java.util.*

interface ProgramRepository : JpaRepository<Program, UUID> {
    fun findAllByUserId(userId: UUID): List<Program>
}
